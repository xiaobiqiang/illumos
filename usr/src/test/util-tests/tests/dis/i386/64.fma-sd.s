/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test ADX related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vfmadd132sd	%xmm0, %xmm1, %xmm2
	vfmadd132sd	(%rax), %xmm1, %xmm2
	vfmadd213sd	%xmm0, %xmm1, %xmm2
	vfmadd213sd	(%rax), %xmm1, %xmm2
	vfmadd231sd	%xmm0, %xmm1, %xmm2
	vfmadd231sd	(%rax), %xmm1, %xmm2
	vfmsub132sd	%xmm0, %xmm1, %xmm2
	vfmsub132sd	(%rax), %xmm1, %xmm2
	vfmsub213sd	%xmm0, %xmm1, %xmm2
	vfmsub213sd	(%rax), %xmm1, %xmm2
	vfmsub231sd	%xmm0, %xmm1, %xmm2
	vfmsub231sd	(%rax), %xmm1, %xmm2
	vfnmadd132sd	%xmm0, %xmm1, %xmm2
	vfnmadd132sd	(%rax), %xmm1, %xmm2
	vfnmadd213sd	%xmm0, %xmm1, %xmm2
	vfnmadd213sd	(%rax), %xmm1, %xmm2
	vfnmadd231sd	%xmm0, %xmm1, %xmm2
	vfnmadd231sd	(%rax), %xmm1, %xmm2
	vfnmsub132sd	%xmm0, %xmm1, %xmm2
	vfnmsub132sd	(%rax), %xmm1, %xmm2
	vfnmsub213sd	%xmm0, %xmm1, %xmm2
	vfnmsub213sd	(%rax), %xmm1, %xmm2
	vfnmsub231sd	%xmm0, %xmm1, %xmm2
	vfnmsub231sd	(%rax), %xmm1, %xmm2
.size libdis_test, [.-libdis_test]
