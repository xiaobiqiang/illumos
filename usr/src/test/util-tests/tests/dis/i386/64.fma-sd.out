    libdis_test:      c4 e2 f1 99 d0     vfmadd132sd %xmm0,%xmm1,%xmm2
    libdis_test+0x5:  c4 e2 f1 99 10     vfmadd132sd (%rax),%xmm1,%xmm2
    libdis_test+0xa:  c4 e2 f1 a9 d0     vfmadd213sd %xmm0,%xmm1,%xmm2
    libdis_test+0xf:  c4 e2 f1 a9 10     vfmadd213sd (%rax),%xmm1,%xmm2
    libdis_test+0x14: c4 e2 f1 b9 d0     vfmadd231sd %xmm0,%xmm1,%xmm2
    libdis_test+0x19: c4 e2 f1 b9 10     vfmadd231sd (%rax),%xmm1,%xmm2
    libdis_test+0x1e: c4 e2 f1 9b d0     vfmsub132sd %xmm0,%xmm1,%xmm2
    libdis_test+0x23: c4 e2 f1 9b 10     vfmsub132sd (%rax),%xmm1,%xmm2
    libdis_test+0x28: c4 e2 f1 ab d0     vfmsub213sd %xmm0,%xmm1,%xmm2
    libdis_test+0x2d: c4 e2 f1 ab 10     vfmsub213sd (%rax),%xmm1,%xmm2
    libdis_test+0x32: c4 e2 f1 bb d0     vfmsub231sd %xmm0,%xmm1,%xmm2
    libdis_test+0x37: c4 e2 f1 bb 10     vfmsub231sd (%rax),%xmm1,%xmm2
    libdis_test+0x3c: c4 e2 f1 9d d0     vfnmadd132sd %xmm0,%xmm1,%xmm2
    libdis_test+0x41: c4 e2 f1 9d 10     vfnmadd132sd (%rax),%xmm1,%xmm2
    libdis_test+0x46: c4 e2 f1 ad d0     vfnmadd213sd %xmm0,%xmm1,%xmm2
    libdis_test+0x4b: c4 e2 f1 ad 10     vfnmadd213sd (%rax),%xmm1,%xmm2
    libdis_test+0x50: c4 e2 f1 bd d0     vfnmadd231sd %xmm0,%xmm1,%xmm2
    libdis_test+0x55: c4 e2 f1 bd 10     vfnmadd231sd (%rax),%xmm1,%xmm2
    libdis_test+0x5a: c4 e2 f1 9f d0     vfnmsub132sd %xmm0,%xmm1,%xmm2
    libdis_test+0x5f: c4 e2 f1 9f 10     vfnmsub132sd (%rax),%xmm1,%xmm2
    libdis_test+0x64: c4 e2 f1 af d0     vfnmsub213sd %xmm0,%xmm1,%xmm2
    libdis_test+0x69: c4 e2 f1 af 10     vfnmsub213sd (%rax),%xmm1,%xmm2
    libdis_test+0x6e: c4 e2 f1 bf d0     vfnmsub231sd %xmm0,%xmm1,%xmm2
    libdis_test+0x73: c4 e2 f1 bf 10     vfnmsub231sd (%rax),%xmm1,%xmm2
